### PR TITLE
Karan's submission

### PR DESCRIPTION
**Pull request by karan01891**

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -2,28 +2,47 @@
...
 
 # Create a program that does the following:
-#     - creates a random integer from 1 - 25
-#     - asks the user to guess what the number is
-#     - validates input is a number
-#     - tells the user if they guess correctly
-#         - if not: tells them too high/low
-#     - only lets the user guess five times
-#         - then ends the program
+#	  - creates a random integer from 1 - 25
```
Comment (danielsgriffin) [inline D04_ex00.py]: Please convert to spaces.

File: D04_ex00.py [hunk 2]
```diff
@@ -2,28 +2,47 @@
...
+#	  - validates input is a number
+#	  - tells the user if they guess correctly
+#		  - if not: tells them too high/low
+#	  - only lets the user guess five times
+#		  - then ends the program
 ################################################################################
 # Imports
-
+import random
+num = random.randint (1,25)
```
Comment (danielsgriffin) [inline D04_ex00.py]: `random.randint (1,25)` => `random.randint(1, 25)`

File: HW04_ch07_ex02.py [hunk 3]
```diff
@@ -20,16 +20,26 @@
 # the last expression it evaluated.
 
 ###############################################################################
-# Imports
-
+# Imports	
+import math
+lst = list()
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: We haven't learned lists yet! But good job.

File: HW04_ch07_ex02.py [hunk 4]
```diff
@@ -20,16 +20,26 @@
...
 # Body
 
+def eval_loop():
+	while True:
+		user_input = input("Enter the string")
+		if user_input == 'done':
+			print (lst[-1])
+			break
+		lst.append(eval(user_input))
+		print(eval(user_input))
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: "return the value of the last expression it evaluated"

File: HW04_ch08_ex05.py [hunk 5]
```diff
@@ -2,3 +2,32 @@
...
+		 #print (convert)
+		 if 97 <= ord(i) <= 122 and convert > 122:
+		 	convert = 97 + (convert - 122 - 1 ) 
+		 	
+
+		 elif 65 <= ord(i) <= 90 and convert > 90:
+		 	convert = 65 + (convert - 90 - 1 ) 
+		 updated_word += chr(convert)
+
+	print(updated_word)
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: "Write a function called rotate_word that takes a string and an integer as parameters, and **returns** a new string that contains the letters from the original string rotated by the given amount."

File: HW04_ch08_ex05.py [hunk 6]
```diff
@@ -2,3 +2,32 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
+
+
+
+def rotate_word(word,number):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Good job.

**Comment by danielsgriffin** [comment]
Good job.

File: D04_ex00.py [hunk 7]
```diff
@@ -2,28 +2,47 @@
...
 
-    print("Hello World!") # Remove this and replace with your function calls
-    
+	guess() # Remove this and replace with your function calls
+	
 
 if __name__ == '__main__':
-    main()
\ No newline at end of file
+	main()
```
Comment (dalmia-nihar) [inline D04_ex00.py]: 1. Got indentation error while running. I think its in the main() function.. 
2. Couldve used try catch to handle exceptional cases
3. The random integer couldve been defined in the main() function

Code logic is solid!

File: HW04_ch07_ex02.py [hunk 8]
```diff
@@ -20,16 +20,26 @@
...
 ###############################################################################
 def main():
-    pass  # Remove this line and uncomment below once eval_loop is defined.
-    # eval_loop()
+	eval_loop()
+
 
 if __name__ == '__main__':
-    main()
+	main()
```
Comment (dalmia-nihar) [inline HW04_ch07_ex02.py]: 1. If you type "done" in the first go itself, the code does not work as expected. A 0 value can be initialised in the list. 

Else the logic looks good!

File: HW04_string2.py [hunk 9]
```diff
@@ -16,9 +16,27 @@
...
+			lst.append ('l')
+			lst.append ('y')
+			s = ''.join(lst)		
+			return s
+		else :
+			s = s + "ing"
+			return s
+	
+	else:
+		return s
```
Comment (dalmia-nihar) [inline HW04_string2.py]: Logic is solid!

File: HW04_string2.py [hunk 10]
```diff
@@ -41,41 +67,73 @@ def not_bad(s):
...
+
+	
+	else:
+		n = int(len(b)/2 +1) 
+		for i in range (0,n,n):
+			b_front = b[i:i+n]
+		for i in range (0,len(b),n):
+			b_back = b[i:i+n]
+
+	return(a_front + b_front + a_back + b_back)
```
Comment (dalmia-nihar) [inline HW04_string2.py]: For loops arent necessary.

The code works well though!

File: HW04_string2.py [hunk 11]
```diff
@@ -30,8 +48,16 @@ def verbing(s):
...
+	# +++your code here+++
+	nott = s.find("not")
+	bad= s.find("bad")
+	bad1 = s.find("bad") + 3
+	if nott < bad :
+		s1 = s[:nott]
+		s2 = s[bad1:]
+		s3 = "good"
+		s = s1 + s3 + s2
+	return(s)
```
Comment (dalmia-nihar) [inline HW04_string2.py]: The check to find if "not" actually exists has not been done. 

The `nott = s.find("not")` will return -1, and `bad= s.find("bad")` will return the location of 'b' in 'bad' .

Meaning the condition 'nott < bad' gets satisfied, and even though 'not .. bad' does not exist, 'good' is inserted.

**Pull request closed without merging**